src/TokenRequestFactory.js

```diff
@@ -1,38 +1,52 @@
 /* @flow */
 
-import { stringify } from 'query-string';
-
+import { URL } from 'universal-url';
 import type { RequestFactory } from './types';
 
 require('fetch-everywhere');
 
 class TokenRequestFactory implements RequestFactory<Request> {
   token: string;
+  baseURL: string;
+  version: string;
 
-  constructor(token: string) {
+  constructor(token: string, baseURL: string, version: string) {
     this.token = token;
+    this.baseURL = baseURL;
+    this.version = version;
+  }
+
+  prefixURI(uri: string): string {
+    const prefix = `/api/${this.version}/`;
+    if (uri.startsWith(prefix)) {
+      return uri;
+    }
+    return `${prefix}${uri}`;
   }
 
-  createRequest(url: string, method?: string = 'GET', body?: Object): Request {
+  createRequest(uri: string, method?: string = 'GET', body?: Object): Request {
+    const url = new URL(this.prefixURI(uri), this.baseURL);
     const headers = {
       Accept: 'application/json',
       'Content-Type': 'application/json; charset=utf-8',
     };
-    const params = {
-      ...body,
-      token: this.token,
-    };
+
+    url.searchParams.append('token', this.token);
 
     if (method === 'GET') {
-      const resolvedURL = `${url}?${stringify(params)}`;
-      return new Request(resolvedURL, {
+      if (body) {
+        Object.entries(body).forEach(entry =>
+          url.searchParams.append(entry[0], String(entry[1])),
```

Comment:
Should this be a `String` or `JSON.stringify` call? I suppose if you're using nested objects on a GET request, that behavior is a bit inconsistent across the internet anyway. 👍